modules/axhal/src/platform/pc_x86/acpi.rs

```diff
@@ -0,0 +1,354 @@
+    fn read_pci_u8(&self, segment: u16, bus: u8, device: u8, function: u8, offset: u16) -> u8 {
+        let paddr = unsafe {
+            ACPI.get_pci_config_regions_addr(segment, bus, device, function)
+                .unwrap()
+        };
+        let vaddr = phys_to_virt(PhysAddr::from(paddr as usize)).as_usize() as *mut u8;
+        let address = unsafe { vaddr.add(offset as usize) };
+        unsafe { address.read_volatile() }
+    }
+
+    fn read_pci_u16(&self, segment: u16, bus: u8, device: u8, function: u8, offset: u16) -> u16 {
+        let paddr = unsafe {
+            ACPI.get_pci_config_regions_addr(segment, bus, device, function)
+                .unwrap()
+        };
+        let vaddr = phys_to_virt(PhysAddr::from(paddr as usize)).as_usize() as *mut u16;
+        let address = unsafe { vaddr.add(offset as usize) };
+        unsafe { address.read_volatile() }
+    }
+
+    fn read_pci_u32(&self, segment: u16, bus: u8, device: u8, function: u8, offset: u16) -> u32 {
+        let paddr = unsafe {
+            ACPI.get_pci_config_regions_addr(segment, bus, device, function)
+                .unwrap()
+        };
+        let vaddr = phys_to_virt(PhysAddr::from(paddr as usize)).as_usize() as *mut u32;
+        let address = unsafe { vaddr.add(offset as usize) };
+        unsafe { address.read_volatile() }
+    }
+
+    fn write_pci_u8(
+        &self,
+        segment: u16,
+        bus: u8,
+        device: u8,
+        function: u8,
+        offset: u16,
+        value: u8,
+    ) {
+        let paddr = unsafe {
+            ACPI.get_pci_config_regions_addr(segment, bus, device, function)
+                .unwrap()
+        };
+        let vaddr = phys_to_virt(PhysAddr::from(paddr as usize)).as_usize() as *mut u8;
+        let address = unsafe { vaddr.add(offset as usize) };
+        unsafe { address.write_volatile(value) }
+    }
+
+    fn write_pci_u16(
+        &self,
+        segment: u16,
+        bus: u8,
+        device: u8,
+        function: u8,
+        offset: u16,
+        value: u16,
+    ) {
+        let paddr = unsafe {
+            ACPI.get_pci_config_regions_addr(segment, bus, device, function)
+                .unwrap()
+        };
+        let vaddr = phys_to_virt(PhysAddr::from(paddr as usize)).as_usize() as *mut u16;
+        let address = unsafe { vaddr.add(offset as usize) };
+        unsafe { address.write_volatile(value) }
+    }
+
+    fn write_pci_u32(
+        &self,
+        segment: u16,
+        bus: u8,
+        device: u8,
+        function: u8,
+        offset: u16,
+        value: u32,
+    ) {
+        let paddr = unsafe {
+            ACPI.get_pci_config_regions_addr(segment, bus, device, function)
+                .unwrap()
+        };
+        let vaddr = phys_to_virt(PhysAddr::from(paddr as usize)).as_usize() as *mut u32;
+        let address = unsafe { vaddr.add(offset as usize) };
+        unsafe { address.write_volatile(value) }
+    }
+}
+
+#[derive(Clone, Debug)]
+struct LocalAllocator;
```

Comment:
This struct is not used?

Reply:
Yes, and i will remove it.
Sorry for this. This struct is a leftover from debugging a new version of acpi.